clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -102,6 +102,9 @@
     /* the client id used to identify this client in requests to the server */
     private final String clientId;
 
+    /* the client software name and commit hash */
+    private final String clientSoftwareNameAndCommit;
```

Comment:
the name suggests it includes commit hash. Do we plan to change this name on every commit?  If so, do we want to config this field automatically?

Reply:
yes, the plan is to for lkc MP to read its resource file and add commit hash to this field, e.g. when TrackingProducer initiates, it will set this config as ||TrackingProducer||linkedin-kafka-clients||<commit_hash>||.

For users who use open source clients directly, this config will be just the default name as "li-oss-kafka-producer-java", and its commit hash will be included in the ApiVersionsRequest.